src/rich-text/plugins/placeholder.ts

```diff
@@ -0,0 +1,40 @@
+import { Fragment } from "prosemirror-model";
+import { Plugin } from "prosemirror-state";
+import { EditorView } from "prosemirror-view";
+
+function showPlaceholder(content: Fragment<any>, textContent: string) {
+    const { firstChild } = content;
+    const { name } = firstChild.type;
+    const allowPlaceholder =
```

Comment:
Hmm... I see where you're going with this, but I'm not entirely sure this is the right approach, since allow-lists can get arbitrarily long. I think we can settle for a slightly worse approach: "Allow any block, so long as the document contains no text". Technically, this will show the placeholder even when the empty document has a styled node opened (such as for a blockquote), but I'm not so sure that's a big deal.

I'll play with it a bit locally and see how I feel about it.

Reply:
> "Allow any block, so long as the document contains no text".

This was my initial approach, but starting off with blockquote (as you mentioned), ordered lists, and unordered lists looks rough.

<details>
<summary>Examples of blockquote, unordered list with placeholder</summary>

<img width="676" alt=" blockquote with placeholder" src="https://user-images.githubusercontent.com/647177/168380432-c5ce4bf2-0da9-4ead-a912-69a6ad80d13b.png">

<img width="682" alt="unordered list with placeholder" src="https://user-images.githubusercontent.com/647177/168380378-d18b924f-0fdf-41df-8138-22cc8a97b514.png">

</details>

Though not ideal, targeting paragraph/heading/markdown only seems specific enough to be safe IMO but I'm open to other approaches.